aztk/internal/cluster_data/node_data.py

```diff
@@ -0,0 +1,166 @@
+import fnmatch
+import io
+import json
+import os
+import yaml
+import zipfile
+from pathlib import Path
+from typing import List
+from aztk.spark import models
+from aztk.utils import constants, file_utils, secure_utils
+from aztk.error import InvalidCustomScriptError
+
+ROOT_PATH = constants.ROOT_PATH
+
+# Constants for node data
+NODE_SCRIPT_FOLDER = "node_scripts"
+CUSTOM_SCRIPT_FOLDER = "custom-scripts"
+CUSTOM_SCRIPT_METADATA_FILE = "custom-scripts.yaml"
+PLUGIN_FOLDER = "plugins"
+
+
+class NodeData:
+    """
+    Class made to bundle data to be uploaded to the node as a zip
+    """
+
+    def __init__(self, cluster_config: models.ClusterConfiguration):
+        self.zip_path = os.path.join(ROOT_PATH, "tmp/node-scripts.zip")
+        self.cluster_config = cluster_config
+        file_utils.ensure_dir(self.zip_path)
+        self.zipf = zipfile.ZipFile(self.zip_path, "w", zipfile.ZIP_DEFLATED)
+
+    def __enter__(self):
```

Comment:
This is nice, and we might want to add it a few other places, but it doesn't look like you used it at all for NodeData?

Reply:
yeah I actually didn't ended up using for node data so could remove it here but there is a few place where this could be useful